Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPIPermissionsCocoa.mm

```diff
@@ -107,6 +110,21 @@
     HashSet<String> permissions, origins;
     parseDetailsDictionary(details, permissions, origins);
 
+    NSString *errorMessage;
+    WebExtension::MatchPatternSet matchPatterns;
+    if (!validatePermissionsDetails(permissions, origins, matchPatterns, @"permissions.remove()", &errorMessage)) {
+        callback->reportError(errorMessage);
+        return;
+    }
+
+    if (!verifyRequestedPermissions(permissions, matchPatterns, @"permissions.remove()", &errorMessage)) {
```

Comment:
Will change this to report an error regardless now that `remove()` returns an error if requesting to remove a permission not defined in the manifest.